cmd/bnsd/x/blueaccount/handler_test.go

```diff
@@ -0,0 +1,1018 @@
+package blueaccount
+
+import (
+	"context"
+	"reflect"
+	"sort"
+	"testing"
+
+	weave "github.com/iov-one/weave"
+	"github.com/iov-one/weave/app"
+	"github.com/iov-one/weave/errors"
+	"github.com/iov-one/weave/gconf"
+	"github.com/iov-one/weave/migration"
+	"github.com/iov-one/weave/store"
+	"github.com/iov-one/weave/weavetest"
+	"github.com/iov-one/weave/weavetest/assert"
+)
+
+func TestUseCases(t *testing.T) {
+	type Request struct {
+		Now         weave.UnixTime
+		Conditions  []weave.Condition
+		Tx          weave.Tx
+		BlockHeight int64
+		WantErr     *errors.Error
+	}
+
+	var (
+		aliceCond   = weavetest.NewCondition()
+		bobCond     = weavetest.NewCondition()
+		charlieCond = weavetest.NewCondition()
+
+		now = weave.UnixTime(1572247483)
+	)
+
+	cases := map[string]struct {
+		Requests  []Request
+		AfterTest func(t *testing.T, db weave.KVStore)
+	}{
+		"configuration can be updated": {
```

Comment:
Please add `success` to the beginning of the test case name so reading the test results will become easier.

Reply:
I agree that adding "success" might help with readability. I do not know where to add it so that it makes sense.
I think you meant prefixing all "successful" tests with "success:". Initially I thought this is a great idea. Now that I am adding it, I am not so sure anymore. The reason is that scenario itself is neither success or failure (unless we talk about the test result). A test scenario in this case consuming messages in order and a certain state at the end is expected. In the process of consuming messages, success of failure is expected as well. I don't think this can be generalized to the whole test description.

In the case of _configuration can be updated_, adding `success:` prefix makes it look weird:

> success: configuration can be updated

Now it looks like I expect the test to succeed. This is true, because I expect ALL tests to succeed :see_no_evil: In my opinion adding `success:` does not provide any information and can only confuse.

What do you think?

Reply:
IMO `success` or `failure` outcome must solely depend on the expected test outcome as you expect this test to **successfully** validate the values returned from config. Basically `happy path`

For the format:
I think it is better to only insert the prefix `success` or `failure` because when a test **explodes** the console output will be `success:_configuration_can_be_updated`. `:` looks weird. And for the sentence structure, I prefer `success update configuration`. Dead simple, no past or future tenses.

Reply:
Ok, now I get it. Makes sense and we can go this direction.

The only complain I have is that `success update configuration` does not sound like a correct phrase. How should I rephrase those descriptions so that they do not sound weird :gb: ?

Reply:
Does it have to be correct and sound English?

Reply:
@willclarktech please save us from this misery 😢